lib/active_record/mysql/enum/quoting.rb

```diff
@@ -8,10 +8,10 @@ module Quoting
       # Quote a symbol as a normal string. This will support quoting of
       # enumerated values.
       def quote(value)
-        if !value.is_a? Symbol
-          __quote_enum(value)
+        if value.is_a? Symbol
+          __quote_enum(value.to_s)
         else
-          ActiveRecord::Base.send(:quote_bound_value, value.to_s)
+          __quote_enum(value)
```

Comment:
I'm confused that `__quote_enum` is just an alias for `quote`. Aren't we at risk for infinite recursion in the `else` case?

Oh, I wonder if this is the old "alias method chain" pattern, where `__quote_enum` is aliased to the version of `quote` that was in place before we override it on line 11? If that's true, this pattern has been superseded by `prepend`, so how about if we use that?

```ruby
class QuoteSymbols
  def quote(value)
    super(value.is_a?(Symbol) ? value.to_s : value)
  end
end

ActiveRecord::ConnectionAdapters::Quoting.prepend QuoteSymbols
```

Reply:
(But also, I wonder if Rails has started allowing symbols to be quoted? That feels like the sort of thing they would do!)

Reply:
You're 100% correct. I removed this code and all the tests passed. Looking at the source code it looks like they handle Symbols:
https://github.com/rails/rails/blob/v6.0.6.1/activerecord/lib/active_record/connection_adapters/abstract/quoting.rb#L228-L229

Reply:
Bingo!